pkg/crc/cluster/kubeadmin_password.go

```diff
@@ -11,47 +13,56 @@ import (
 	"github.com/code-ready/crc/pkg/crc/constants"
 	"github.com/code-ready/crc/pkg/crc/logging"
 	"github.com/code-ready/crc/pkg/crc/oc"
-	crcos "github.com/code-ready/crc/pkg/os"
 	"golang.org/x/crypto/bcrypt"
 )
 
-// UpdateKubeAdminUserPassword does following
-// - Create and put updated kubeadmin password to ~/.crc/machine/crc/kubeadmin-password
-// - Update the htpasswd secret
-func UpdateKubeAdminUserPassword(ocConfig oc.Config) error {
+// GenerateKubeAdminUserPassword creates and put updated kubeadmin password to ~/.crc/machine/crc/kubeadmin-password
+func GenerateKubeAdminUserPassword() error {
+	logging.Infof("Generating new password for the kubeadmin user")
 	kubeAdminPasswordFile := constants.GetKubeAdminPasswordPath()
-	if crcos.FileExists(kubeAdminPasswordFile) {
-		logging.Debugf("kubeadmin password has already been updated")
-		return nil
+	kubeAdminPassword, err := GenerateRandomPasswordHash(23)
+	if err != nil {
+		return fmt.Errorf("Cannot generate the kubeadmin user password: %w", err)
 	}
+	return ioutil.WriteFile(kubeAdminPasswordFile, []byte(kubeAdminPassword), 0600)
+}
 
-	logging.Infof("Generating new password for the kubeadmin user")
-
-	kubeAdminPassword, err := GenerateRandomPasswordHash(23)
+// UpdateKubeAdminUserPassword updates the htpasswd secret
+func UpdateKubeAdminUserPassword(ocConfig oc.Config) error {
+	kubeAdminPassword, err := GetKubeadminPassword()
 	if err != nil {
 		return fmt.Errorf("Cannot generate the kubeadmin user password: %w", err)
```

Comment:
This error need to be updated, it is not any more issue with generation but reading the kubeconfig file with password.